java-client/weather-server-example/src/main/java/io/deephaven/client/examples/ui/Ui.java

```diff
@@ -0,0 +1,262 @@
+    private SessionImpl session;
+
+    // Connection related items both for GRPC and Flight.
+    private final ScheduledExecutorService flightScheduler = Executors.newScheduledThreadPool(8);
+    private ManagedChannel managedChannel;
+    private BarrageSupport support;
+
+
+    private BarrageTable statsTable;
+
+    public Ui() {
+        // Start the LTM. This module is responsible for deterministically handling table updates
+        // In other words, it drives the 'ticking' of tables.
+        UpdateGraphProcessor.DEFAULT.start();
+
+        connectButton.addActionListener(this::doConnect);
+        addButton.addActionListener(this::onAdd);
+
+        Runtime.getRuntime().addShutdownHook(new Thread(this::onShutdown));
+    }
+
+    private void onAdd(final ActionEvent ev) {
+        // First we'll just ship the request to the server and wait for the response.
+        // If we get an Ack (true) We'll stuff this thing in the list. This is crude on purpose --
+        // We don't actually care about having complex logic to track places, only that we can submit them.
+        final String place = locationField.getText();
+        if (place == null || place.isEmpty()) {
+            return;
+        }
+
+        try (final ConsoleSession console = session.console("python").get()) {
+            SwingUtilities.invokeLater(() -> statusLabel.setText("Attempting to add " + place));
+            final Changes c = console.executeCode("beginWatch(\"" + place + "\")");
+            if (c.errorMessage().isPresent()) {
+                SwingUtilities.invokeLater(
+                        () -> statusLabel.setText("Error adding " + place + " -> " + c.errorMessage().get()));
+            } else {
+                SwingUtilities.invokeLater(() -> statusLabel.setText("Added " + place));
+            }
+        } catch (ExecutionException | InterruptedException | TimeoutException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Connect to the Deephaven Community Core server via Arrow FLight and fetch the relevant tables for this example.
+     *
+     * @param ev
+     */
+    private void doConnect(final ActionEvent ev) {
+        // This bit is an oddity of the Swing UI implementation -- we just ensure we are not
+        // executing potentially blocking I/O on the thread that handles UI management & drawing.
+        if (SwingUtilities.isEventDispatchThread()) {
+            flightScheduler.schedule(() -> doConnect(ev), 0, TimeUnit.SECONDS);
+            return;
+        }
+
+        // Do we have everything we need to know to connect? Host / port?
+        final String host = hostField.getText();
+        if (host == null || host.isEmpty()) {
+            return;
+        }
+
+        // Close any pre-existing session and wait for it to terminate
+        if (session != null) {
+            try {
+                support.close();
+                session.close();
+                managedChannel.shutdownNow();
```

Comment:
shutdown process should look like java exec service pattern, roughly: shutdown, awaitTermination, shutdownNow, awaitTermination, this lets existing work finish while rejecting new work, and after the timeout, murder even pending work

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html